macros/rp-connect-components.js

```diff
@@ -1,6 +1,117 @@
+    
+    // Check if the supportFilter element exists
+    const supportFilterElement = document.querySelector('#supportFilter');
+    const supportFilter = supportFilterElement 
+        ? Array.from(supportFilterElement.selectedOptions).map(option => option.value)
+        : [];
+
+    const table = document.getElementById('componentTable');
+    const trs = table.getElementsByTagName('tr');
+
+    for (let i = 1; i < trs.length; i++) {
+        const row = trs[i];
+        const nameTd = row.querySelector('td[id^="componentName-"]');
+        const typeTd = row.querySelector('td[id^="componentType-"]');
+        const supportTd = row.querySelector('td[id^="componentSupport-"]');
+        const typeDropdown = typeTd ? typeTd.querySelector('.type-dropdown') : null;
+
+        if (nameTd && typeTd) {
+            const nameText = nameTd.textContent.trim().toLowerCase();
+            const typeText = typeTd.textContent.trim().toLowerCase().split(', ').map(item => item.trim());
+            const supportText = supportTd ? supportTd.textContent.trim().toLowerCase() : '';
+
+            // Determine if the row should be shown
+            const showRow =
+                ((!nameInput || nameText.includes(nameInput)) &&
+                (typeFilter.length === 0 || typeFilter.some(value => typeText.includes(value))) &&
+                (!supportTd || supportFilter.length === 0 || supportFilter.some(value => supportText.includes(value)))
+                );
+
+            row.style.display = showRow ? '' : 'none';
+
+            if (showRow && typeFilter.length > 0 && typeDropdown) {
+                const matchingOption = Array.from(typeDropdown.options).find(option => 
+                    typeFilter.includes(option.text.toLowerCase())
+                );
+                if (matchingOption) {
+                    typeDropdown.value = matchingOption.value;
+                    updateComponentUrl(typeDropdown, false);
+                }
+            }
+        } else {
+            row.style.display = 'none'; // Hide row if essential cells are missing
+        }
+    }
+}
+
+  const capitalize = s => s && s[0].toUpperCase() + s.slice(1);
+
+  function processConnectors(parsedData) {
+    return parsedData.data.reduce((connectors, row) => {
+      const { connector, commercial_name, type, support_level, is_licensed, url } = row;
+      if (!connectors[connector]) {
+        connectors[connector] = {
+          types: new Map(),
+          supportLevels: new Map(),
+          isLicensed: is_licensed === 'y' ? 'Yes' : 'No',
+          urls: new Set()
+        };
+      }
+      connectors[connector].types.set(capitalize(type), url);
+      if (url) connectors[connector].urls.add(url);
+      if (!connectors[connector].supportLevels.has(support_level)) {
+        connectors[connector].supportLevels.set(support_level, new Set());
+      }
+      connectors[connector].supportLevels.get(support_level).add(commercial_name);
```

Comment:
I see you're relying on there being multiple rows for a `connector` and the `commercial_name` being the specific flavor (driver, cache etc..). But Ash's CSV does not follow this pattern. We'll need to update this logic to use whatever Ash has here: https://github.com/redpanda-data/connect/pull/2827#discussion_r1743287405

For example, the support for each `cache` output target should be fetched from the `cache` connector types.

`cache` connectors:

```csv
cache                   ,output    ,cache                   ,certified  ,n          ,y     ,y
cache                   ,processor ,cache                   ,certified  ,n          ,y     ,y
```

The support levels for each cache type is defined in the connectors with type `cache`:

```
gcp_cloud_storage       ,cache     ,gcp_cloud_storage       ,community  ,n          ,n     ,n
lru                     ,cache     ,lru                     ,community  ,n          ,y     ,y
```